### PR TITLE
Always test AMGX in parallel

### PR DESCRIPTION
**Pull request by masterleinad**
We should also run this test when the number of devices is not equal to 2.

### CODE REVIEW

File: tests/test_amgx_direct_solver.cu [hunk 1]
```diff
@@ -122,85 +122,81 @@ BOOST_AUTO_TEST_CASE(amgx_2_procs)
   cudaError_t cuda_error_code = cudaGetDeviceCount(&n_devices);
   mfmg::ASSERT_CUDA(cuda_error_code);
   int comm_size = dealii::Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD);
-  if ((n_devices == 2) && (comm_size == 2))
+  if (comm_size == 2)
```
Comment (dalg24) [inline tests/test_amgx_direct_solver.cu]: Do we want that `if` at all then?

**Comment by masterleinad** [comment]
Yes, I do.

File: tests/test_amgx_direct_solver.cu (hunk 1 above)
Comment (dalg24) [reply tests/test_amgx_direct_solver.cu]: Ping

Comment (masterleinad) [reply tests/test_amgx_direct_solver.cu]: I don't see a reason why the test should not work with just one `MPI` process. Let's see what `CI` says in another pull request.

Comment (masterleinad) [reply tests/test_amgx_direct_solver.cu]: See #206.

**Pull request merged**